prime-router/src/testIntegration/kotlin/serializers/Hl7SerializerIntegrationTests.kt

```diff
@@ -667,4 +658,23 @@ NTE|1|L|This is a final comment|RE"""
         assertThat(msh43).isEqualTo("correctText-YES!")
         assertThat(msh10).isEqualTo("yeah/success")
     }
+
+    @Test
+    fun `test NTE Source`() {
+        val parser = context.pipeParser
+
+        val uploadStream = File("./src/test/unit_test_files/csv-upload-test.csv").inputStream()
+        val uploadSchema = "upload-covid-19"
+        val sender = Sender("default", "upload", Sender.Format.CSV, "covid-19", CustomerStatus.TESTING, uploadSchema)
+        val testReport = csvSerializer.readExternal(uploadSchema, uploadStream, TestSource, sender).report
+        val output = serializer.buildMessage(testReport, 0)
+        val hapiMsg = parser.parse(output.toString())
+        val terser = Terser(hapiMsg)
+        val nte21 = terser.get("/PATIENT_RESULT/ORDER_OBSERVATION/OBSERVATION/NTE(0)-2-1") ?: ""
+        val nte22 = terser.get("/PATIENT_RESULT/ORDER_OBSERVATION/OBSERVATION/NTE(0)-2-2")
+
+        if (nte21.isNotEmpty()) {
```

Comment:
This if statement should not be here.  A test should always assert.  If the test data makes this not empty then replace with assertThat(nte21).isNotEmpty or not blank.